client/helpers/navigate.js

```diff
@@ -38,7 +38,15 @@ function encode_path(path){
 }
 
 export function prepare(path){
-    return encodeURIComponent(decodeURIComponent(path)); // to send our url correctly without using directly '/'
+    // to send our url correctly without using directly '/'
+    let decoded = '';
+    try {
```

Comment:
I'd rather avoid using try/catch in this context and make sure the code calling prepare doens't throw in the first place.

Reply:
It should be same to skip the try/catch and simply replace the `%` on decode as there shouldn't be any URL encoding at this point.

Reply:
sounds good if that works out